app/scripts/modules/core/src/application/application.model.ts

```diff
@@ -80,21 +81,46 @@ export class Application {
    */
   public activeState: ApplicationDataSource = null;
 
-  // Since active state changes don't use $scope, we can just make the Subject public
+  public activeDataSource$ = new ReplaySubject<ApplicationDataSource>(1);
+  /** @deprecated use activeDataSource$ */
   public activeStateChangeStream: Subject<any> = new Subject();
 
   private refreshStream: Subject<any> = new Subject();
-
   private refreshFailureStream: Subject<any> = new Subject();
 
+  // Stream of fetch status
+  public status$: Observable<IFetchStatus>;
+
   private dataLoader: Subscription;
 
-  constructor(
-    private applicationName: string,
-    private scheduler: any,
-    private $q: IQService,
-    private $log: ILogService,
-  ) {}
+  constructor(private applicationName: string, private scheduler: any, dataSourceConfigs: IDataSourceConfig[]) {
+    dataSourceConfigs.forEach(config => this.addDataSource(config));
+    this.status$ = Observable.combineLatest(this.dataSources.map(ds => ds.status$)).map(statuses =>
+      this.getDerivedApplicationStatus(statuses),
+    );
+  }
+
+  private getDerivedApplicationStatus(statuses: IFetchStatus[]): IFetchStatus {
+    const ERROR_STATUS = statuses.some(({ status }) => status === 'ERROR') ? 'ERROR' : undefined;
+    const FETCHING_STATUS = statuses.some(({ status }) => status === 'FETCHING') ? 'FETCHING' : undefined;
+    const FETCHED_STATUS = statuses.some(({ status }) => status === 'FETCHED') ? 'FETCHED' : undefined;
+
+    const rolledUpStatus: IFetchStatus['status'] = ERROR_STATUS || FETCHING_STATUS || FETCHED_STATUS;
```

Comment:
You had a `NOT_INITIALIZED` state at one point. Not sure if that was missed or not needed.